### PR TITLE
Add functionality to capture and suppress stdio

### PR DESCRIPTION
**Pull request by joac**
This change introduces two new config settings:
- `captureOutput` that stores all stdio for the current scenario and is displayed when an error happens
- `suppressOutput` that prevents strings being printed on stdio (interleaving with the dots)

image

### CODE REVIEW

File: src/output.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  const originalWrite = target.write;
+  return {
+    capturedData,
+    intercept: () => {
+      target.write = (a: string) => {
+        if (capture) {
+          capturedData.push(a);
+        }
+        if (!suppress) {
+          originalWrite.bind(target)(a);
```
Comment (pzavolinsky) [inline src/output.ts:33]: do `originalWrite.call(target, a)` instead to avoid creating a new function only for this execution

File: src/test.ts [hunk 2]
```diff
@@ -149,44 +142,126 @@ const fn: SetupFn = ({ getCtx, Given, onTearDown, setCtx, Then, When }) => {
...
+      setCtx('test-suite-stdout', output.stdout);
+      setCtx('test-suite-stderr', output.stderr);
+    } catch (error) {
+      // Test suite can fail
+      setCtx('test-suite-stdout', error.stdout);
+      setCtx('test-suite-stderr', error.stderr);
+    }
+
+    onTearDown(async () => {
+      testDir && rmdir(testDir, { recursive: true });
```
Comment (pzavolinsky) [inline src/test.ts:237]: use an `if`

File: src/output.ts (hunk 1 above)
Comment (joac) [reply src/output.ts]: Fixed

File: src/test.ts (hunk 2 above)
Comment (joac) [reply src/test.ts]: :+1:  Done

**Pull request merged**